cmd/generate/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configtempov1alpha1 "github.com/os-observability/tempo-operator/apis/config/v1alpha1"
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+	"github.com/os-observability/tempo-operator/internal/manifests"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+}
+
+func loadSpec(path string) (v1alpha1.Microservices, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return v1alpha1.Microservices{}, err
+	}
+	defer file.Close()
+
+	spec := v1alpha1.Microservices{}
+	decoder := yaml.NewYAMLOrJSONDecoder(file, 8192)
```

Comment:
`8192` is a memory limiation? ^^

Reply:
Docs say `bufferSize determines how far into the stream the decoder will look to figure out whether this is a JSON stream (has whitespace followed by an open brace).`
Imho it shouldn't have to look very far to determine if the input file is a JSON or YAML.

Reply:
Can we put this number in a constant? and a comment indicating what it is?

Reply:
Done.